http/obmc_shell.hpp

```diff
@@ -0,0 +1,250 @@
+            streamFileDescriptor.assign(ttyFileDescriptor);
+            doWrite();
+            doRead();
+        }
+    }
+
+    void doWrite()
+    {
+        if (session == nullptr)
+        {
+            BMCWEB_LOG_DEBUG << "session is closed";
+            return;
+        }
+        if (doingWrite)
+        {
+            BMCWEB_LOG_DEBUG << "Already writing.  Bailing out";
+            return;
+        }
+
+        if (inputBuffer.empty())
+        {
+            BMCWEB_LOG_DEBUG << "inputBuffer empty.  Bailing out";
+            return;
+        }
+
+        doingWrite = true;
+        streamFileDescriptor.async_write_some(
+            boost::asio::buffer(inputBuffer.data(), inputBuffer.size()),
+            [this, self(shared_from_this())](boost::beast::error_code ec,
+                                             std::size_t bytesWritten) {
+            BMCWEB_LOG_DEBUG << "Wrote " << bytesWritten << "bytes";
+            doingWrite = false;
+            inputBuffer.erase(0, bytesWritten);
+            if (ec == boost::asio::error::eof)
+            {
+                session->close("ssh socket port closed");
+                return;
+            }
+            if (ec)
+            {
+                session->close("Error in writing to processSSH port");
+                BMCWEB_LOG_ERROR << "Error in ssh socket write " << ec;
+                return;
+            }
+            doWrite();
+            });
+    }
+
+    void doRead()
+    {
+        if (session == nullptr)
+        {
+            BMCWEB_LOG_DEBUG << "session is closed";
+            return;
+        }
+        streamFileDescriptor.async_read_some(
+            boost::asio::buffer(outputBuffer.data(), outputBuffer.size()),
+            [this, self(shared_from_this())](
+                const boost::system::error_code& ec, std::size_t bytesRead) {
+            BMCWEB_LOG_DEBUG << "Read done.  Read " << bytesRead << " bytes";
+            if (session == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "session is closed";
+                return;
+            }
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "Couldn't read from ssh port: " << ec;
+                session->close("Error in connecting to ssh port");
+                return;
+            }
+            std::string_view payload(outputBuffer.data(), bytesRead);
+            session->sendBinary(payload);
+            doRead();
+            });
+    }
+
+    // this has to public
+    std::string inputBuffer;
+
+  private:
+    crow::websocket::Connection* session;
+    boost::asio::posix::stream_descriptor streamFileDescriptor;
+    bool doingWrite{false};
+    int ttyFileDescriptor{0};
+    pid_t pid{0};
+
+    std::array<char, 4096> outputBuffer{};
+};
+
+static std::map<crow::websocket::Connection*, std::shared_ptr<Handler>>
+    mapHandler;
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(app, "/bmc-console")
+        .privileges({{"ConfigureManager"}})
```

Comment:
We need this to service at some point

Reply:
https://github.com/ibm-openbmc/bmcweb/blob/1030/http/obmc_shell.hpp#L177

Reply:
https://github.com/ibm-openbmc/bmcweb/blob/1030/http/obmc_shell.hpp#L185

Reply:
https://github.com/ibm-openbmc/bmcweb/commit/48b082d25e6727dbdfe8e03c08a6c160ebebc388